nstar/src/main.rs

```diff
@@ -71,61 +82,73 @@ async fn print_response<W: Write>(mut out: &mut W, response: Response, json: boo
+                        } else {
+                            break 'outer;
+                        };
+                        match parse_prompt(input) {
+                            Ok(PromptCommand::Prompt(cmd)) => match cmd {
+                                Prompt::Help => print_help(&mut terminal)?,
+                                Prompt::Quit => {
+                                    writeln!(&mut terminal, "bye!")?;
+                                    break 'outer
+                                },
+                            },
+                            Ok(PromptCommand::Northstar(cmd)) => {
+                                match send_request(&mut client, &cmd).await {
+                                    Ok(response) => print_response(&mut terminal, response, opt.json).await?,
+                                    // break the loop and try to reconnect
+                                    Err(api::client::Error::Stopped) => break,
+                                    Err(e) => writeln!(&mut terminal, "{}: {}", "client error".red(), e)?,
+                                };
+                            },
+                            Err(e) => writeln!(&mut terminal, "{}", e.message)?
+                        };
                     }
-                }
-                input = input.next() => {
-                    let input: &str = if input.is_some() {
-                        input.as_deref().unwrap()
-                    } else {
-                        break 'outer;
-                    };
-                    match parse_prompt(input) {
-                        Ok(PromptCommand::Prompt(cmd)) => match cmd {
-                            Prompt::Help => print_help(&mut terminal)?,
-                            Prompt::Quit => break 'outer,
-                        },
-                        Ok(PromptCommand::Northstar(cmd)) => {
-                            match send_request(&mut client, &cmd).await {
-                                Ok(response) => print_response(&mut terminal, response, opt.json).await?,
-                                // break the loop and try to reconnect
-                                Err(api::client::Error::Stopped) => break,
-                                Err(e) => writeln!(&mut terminal, "{}: {}", "client error".red(), e)?,
-                            };
-                        },
-                        Err(e) => writeln!(&mut terminal, "{}", e.message)?
-                    };
                 }
             }
+
+            // Try to reconnect
+            writeln!(terminal, "Disconnected, trying to reconnect")?;
+            client = try_connect(&mut std::io::stdout(), &url).await?;
         }
+        Ok(())
+    });
 
-        writeln!(terminal, "Disconnected")?;
-        time::sleep(time::Duration::from_secs(1)).await;
+    // Shutdown the runtime
+    // Also stops the terminal blocked reading from stdin
+    runtime.shutdown_background();
```

Comment:
Why is this needed when exiting anyways?

Reply:
This is needed because the terminal tasks that reads from `stdin` is blocked waiting for input at this point.

Reply:
The documentation os `shutdown_backgroud` explains that the tasks are just dropped and leaked. Feels a little bit hacky. Is there a way to unblock the terminal task and terminate gracefully?

Reply:
I tried a couple of approaches and nothing seems to stop the blocked task. This is the only thing that "worked" so far.